platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapView.java

```diff
@@ -919,26 +963,21 @@ private void setZoom(double zoomLevel, @Nullable PointF focalPoint, @NonNull Tra
 
   private static class MapCallback implements OnMapChangedListener {
 
-    private final MapboxMap mapboxMap;
+    private MapboxMap mapboxMap;
     private final List<OnMapReadyCallback> onMapReadyCallbackList = new ArrayList<>();
     private boolean initialLoad = true;
 
-    MapCallback(MapboxMap mapboxMap) {
+    void attachMapboxMap(MapboxMap mapboxMap) {
       this.mapboxMap = mapboxMap;
     }
 
     @Override
     public void onMapChanged(@MapChange int change) {
       if (change == DID_FINISH_LOADING_STYLE && initialLoad) {
         initialLoad = false;
-        new Handler().post(new Runnable() {
```

Comment:
afaik, we posted  to the end of message queue to make sure the view was measured (to be able to initialize all components with correct width/height) . Now with this PR, we postpone map creation and thus style loading to after onResume (when the view is toggled visible). Can't really think of another reason why we needed that post there in the first place?